sauron/sauron.py

```diff
@@ -101,16 +101,31 @@ def getchaininfo(plugin, **kwargs):
 @plugin.method("getrawblockbyheight")
 def getrawblock(plugin, height, **kwargs):
     blockhash_url = "{}/block-height/{}".format(plugin.api_endpoint, height)
-
     blockhash_req = fetch(blockhash_url)
-    block_req = fetch("{}/block/{}/raw".format(plugin.api_endpoint,
-                                               blockhash_req.text))
-    if blockhash_req.status_code != 200 or block_req.status_code != 200:
+    if blockhash_req.status_code != 200:
         return {
             "blockhash": None,
             "block": None,
         }
 
+    block_url = "{}/block/{}/raw".format(plugin.api_endpoint, blockhash_req.text)
+    while True:
+        block_req = fetch(block_url)
+        if block_req.status_code != 200:
+            return {
+                "blockhash": None,
+                "block": None,
+            }
+        # We may download partial/incomplete files for Esplora. Best effort to
+        # not crash lightningd by sending an invalid (trimmed) block.
+        content_len = block_req.headers.get("Content-length")
+        if content_len is None:
```

Comment:
You could change this block so it is more readable (for me at least). Also its advisable to sleep a certain timeout before retrying:
```python
if content_len != None and int(content_len) != len(block_req.content):
    plugin.log("Esplora gave us an incomplete block, retrying", level="error")
    time.sleep(3)
    continue
```

Reply:
concept ack, I think a cool think to have here is a complete porting of my backend implementation in java 

https://github.com/clightning4j/btcli4j/blob/81a6616839a14d20834f124639c01d931c350af8/src/main/kotlin/io/vincenzopalazzo/btcli4j/util/HttpRequestFactory.kt#L117-L127

It is cool to make an increasing waiting time here because the problem could not be the esplora API but the local network.

In my code, if the error happens after 4 times with the increasing waiting time, now the plugin has my permission to die.

Reply:
I find it more readable this way :p Added a sleep though!